samples/socket-api/wasm-src/timeout_server.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2022 Amazon.com Inc. or its affiliates. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <arpa/inet.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/time.h>
+#include <unistd.h>
+#ifdef __wasi__
+#include <wasi_socket_ext.h>
+#endif
+
+int
+main(int argc, char *argv[])
+{
+    int socket_fd;
+    int client_socket_fd;
+    struct sockaddr_in addr = { 0 };
+    int addrlen = sizeof(addr);
+    int bool_opt = 1;
+
+    addr.sin_family = AF_INET;
+    addr.sin_port = htons(1234);
+    addr.sin_addr.s_addr = htonl(INADDR_ANY);
+    if ((socket_fd = socket(AF_INET, SOCK_STREAM, 0)) == -1) {
+        perror("Create socket failed");
+        return EXIT_FAILURE;
+    }
+
+    if (setsockopt(socket_fd, SOL_SOCKET, SO_REUSEADDR, &bool_opt,
+                   sizeof(bool_opt))
+        == -1) {
+        perror("Failed setting SO_REUSEADDR");
+        close(socket_fd);
+        return EXIT_FAILURE;
```

Comment:
Same as above, had better define a label `fail` to handle the failures here and L42/L48/L56